subsys/bluetooth/mesh/cdb.c

```diff
@@ -858,8 +868,7 @@ void bt_mesh_cdb_node_del(struct bt_mesh_cdb_node *node, bool store)
 		update_cdb_node_settings(node, false);
 	}
 
-	node->addr = BT_MESH_ADDR_UNASSIGNED;
-	memset(node->dev_key, 0, sizeof(node->dev_key));
+	atomic_set_bit(node->flags, BT_MESH_CDB_NODE_REMOVED);
```

Comment:
I wonder if we add `CONFIG_BT_MESH_CDB_NODE_COUNT` and then remove all them, will it be possible to add one more entry? I am thinking if cdb should just start from the last free address that is address of the element of the last provisioned node + 1 in `find_lowest_free_addr`? Then when it hits the limit of unicast addresses, it blocks adding a new node. If IV Update happens, it starts iterating again from address 1.

Reply:
Meaning that if someone does `bt_mesh_cdb_node_alloc(x, 0x7fff, x, x)` they now can't auto-assign any addresses until after IV-update? It is possible to allocate a node with a specific address and then remove this, the addresses allocated are not guaranteed to be contiguous.

Reply:
> Meaning that if someone does `bt_mesh_cdb_node_alloc(x, 0x7fff, x, x)` and then delete this node, they now can't auto-assign any addresses until after IV-update?

This could be an API limitation. Though in theory could be solved by e.g. adding a map of used address. 

I tried this in shell and I wasn't able to add a 9th node after adding 8 and removing some of them:
```
uart:~$ mesh cdb-create
uart:~$ mesh cdb-node-add dddd 0 1 0
Added node 0x0000
uart:~$ mesh cdb-node-add dddd 0 1 0
Added node 0x0000
uart:~$ mesh cdb-node-add dddd 0 1 0
Added node 0x0000
uart:~$ mesh cdb-node-add dddd 0 1 0
Added node 0x0000
uart:~$ mesh cdb-node-add dddd 0 1 0
Added node 0x0000
uart:~$ mesh cdb-node-add dddd 0 1 0
Added node 0x0000
uart:~$ mesh cdb-node-add dddd 0 1 0
Added node 0x0000
uart:~$ mesh cdb-node-add dddd 0 1 0
Added node 0x0000
uart:~$ mesh cdb-node-add dddd 0 1 0
Failed to allocate node
uart:~$ mesh cdb-node-del 8
Deleted node 0x0008
uart:~$ mesh cdb-node-add dddd 0 1 0
Failed to allocate node
uart:~$ mesh cdb-show
Mesh Network Information
========================
Address  Elements  Flags  UUID                              DevKey
0x0001   1         --     dddd0000000000000000000000000000  b798ec90decbe9f46f1f936676d6eb7b
0x0002   1         --     dddd0000000000000000000000000000  e9425870b161f4a9d986ed44ef42c2c6
0x0003   1         --     dddd0000000000000000000000000000  7c38a677c6e175b834eebe87bcd9c8f2
0x0004   1         --     dddd0000000000000000000000000000  fe68f30debaa57f779c8de580e27e598
0x0005   1         --     dddd0000000000000000000000000000  dc28fbcc30ac7ba254105cdb90fb3b79
0x0006   1         --     dddd0000000000000000000000000000  ffb09a84e00395a7de23347ba48e1c3b
0x0007   1         --     dddd0000000000000000000000000000  bfb570cc9ada982598c288c1298e1437
0x0008   1         -R     dddd0000000000000000000000000000  85b6681c4731bc6576dbfa1e33c02d19
> Total nodes: 8
---
NetIdx  NetKey
0x000   bee7eaaf01d6ca347622c106c60f9a79
> Total subnets: 1
---
NetIdx  AppIdx  AppKey
> Total app-keys: 0
```

Reply:
Yep you are correct of course, I just think this limitation is more sensible than breaking the auto-address function when a high-address node is added manually? Adding a map of used addresses would of course also contain a limit on the number of slots in this map, so that wouldn't "solve" the problem either.

Reply:
In my opinion it is more sensible to start from the last allocated address + number of elements + 1 than limiting number of possibly added nodes to `CONFIG_BT_MESH_CDB_NODE_COUNT`. I'm not sure what is the use case for allocating node with address `0x7fff`. But to workaround this problem, we could count this address only after such node gets deleted.

Reply:
Hm. Deferring the update of free address to after deletion makes it less of an issue. I think I can agree now